pkg/api/object.go

```diff
@@ -221,11 +220,128 @@ func (h *Handler) Listbuckets(w http.ResponseWriter, r *http.Request) {
 	FormatJsonResponse(w, sortedBuckets)
 }
 
-func handleConnectionLookupFailure(err error, action string, w http.ResponseWriter) {
-	logger.Errorf("failed to get %s: %+v", action, err)
-	if util.IsEtcdKeyNotFound(err) {
-		w.WriteHeader(http.StatusNotFound)
-	} else {
+// CreateBucket creates a bucket in the object store in this cluster.
+// POST
+// /objectstore/buckets/{BUCKET_NAME}
+func (h *Handler) CreateBucket(w http.ResponseWriter, r *http.Request) {
+	var bucket model.ObjectBucket
+
+	err := json.NewDecoder(r.Body).Decode(&bucket)
+	if err != nil {
+		logger.Errorf("Error parsing bucket: %+v", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if bucket.Name == "" {
+		logger.Errorf("Error: bucket name cannot be empty")
+		w.WriteHeader(http.StatusUnprocessableEntity)
+		w.Write([]byte("bucket name cannot be empty"))
+		return
+	}
+
+	if bucket.Owner == "" {
+		logger.Errorf("Error: bucket owner cannot be empty")
+		w.WriteHeader(http.StatusUnprocessableEntity)
+		w.Write([]byte("bucket owner cannot be empty"))
+		return
+	}
+
+	createdBucket, err := rgw.CreateBucket(h.context, bucket, h.config.ClusterHandler.GetClusterInfo)
+	if err != nil {
+		logger.Errorf("Error creating bucket: %+v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	FormatJsonResponse(w, createdBucket)
+}
+
+// GetBucket gets the bucket from the object store in this cluster.
+// GET
+// /objectstore/users/{USER_ID}
+func (h *Handler) GetBucket(w http.ResponseWriter, r *http.Request) {
+	bucketName := mux.Vars(r)["bucketName"]
+
+	user, rgwError, err := rgw.GetBucket(h.context, bucketName, h.config.ClusterHandler.GetClusterInfo)
+	if err != nil {
+		logger.Errorf("Error getting user (%s): %+v", bucketName, err)
+
+		if rgwError == rgw.RGWErrorNotFound {
+			w.WriteHeader(http.StatusNotFound)
+		} else {
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+		return
+	}
+
+	FormatJsonResponse(w, user)
+}
+
+// GetBucketACL gets the ACL of the bucket in the object store in this cluster.
+// GET
+// /objectstore/bucketACLs/{BUCKET_NAME}
+func (h *Handler) GetBucketACL(w http.ResponseWriter, r *http.Request) {
+	bucketName := mux.Vars(r)["bucketName"]
+
+	acl, err := rgw.GetBucketACL(h.context, bucketName, h.config.ClusterHandler.GetClusterInfo)
+	if err != nil {
+		logger.Errorf("Error getting bucket ACL: %+v", err)
+		w.WriteHeader(http.StatusInternalServerError)
```

Comment:
At this point there is no mechanism for distinguishing between not found and other errors. This should change once we have an the code written that backs this in ceph.